samza-test/src/test/java/org/apache/samza/test/operator/StreamApplicationIntegrationTestHarness.java

```diff
@@ -0,0 +1,253 @@
+   *
+   * The default poll time out is determined by {@link #POLL_TIMEOUT_MS} and the number of empty polls are
+   * determined by {@link #numEmptyPolls}
+   *
+   * @param topics the list of topics to consume from
+   * @param threshold the number of messages to consume
+   * @return the list of {@link ConsumerRecord}s whose size can be atmost {@param threshold}
+   */
+  public List<ConsumerRecord<String, String>> consumeMessages(Collection<String> topics, int threshold) {
+    int emptyPollCount = 0;
+    List<ConsumerRecord<String, String>> recordList = new ArrayList<>();
+    consumer.subscribe(topics);
+
+    while (emptyPollCount < numEmptyPolls && recordList.size() < threshold) {
+      ConsumerRecords<String, String> records = consumer.poll(POLL_TIMEOUT_MS.toMillis());
+      if (!records.isEmpty()) {
+        Iterator<ConsumerRecord<String, String>> iterator = records.iterator();
+        while (iterator.hasNext() && recordList.size() < threshold) {
+          ConsumerRecord record = iterator.next();
+          recordList.add(record);
+          emptyPollCount = 0;
+        }
+      } else {
+        emptyPollCount++;
+      }
+    }
+    return recordList;
+  }
+
+  /**
+   * Executes the provided {@link StreamApplication} as a {@link org.apache.samza.job.local.ThreadJob}. The
+   * {@link StreamApplication} runs in its own separate thread.
+   *
+   * @param streamApplication the application to run
+   * @param appName the name of the application
+   * @param overriddenConfigs configs to override
+   */
+  public void runApplication(StreamApplication streamApplication, String appName, Config overriddenConfigs) {
+
+    Map<String, String> configs = new HashMap<>();
+    configs.put("job.factory.class", "org.apache.samza.job.local.ThreadJobFactory");
+    configs.put("job.name", appName);
+    configs.put("app.class", streamApplication.getClass().getCanonicalName());
+    configs.put("serializers.registry.json.class", "org.apache.samza.serializers.JsonSerdeFactory");
+    configs.put("serializers.registry.string.class", "org.apache.samza.serializers.StringSerdeFactory");
+    configs.put("systems.kafka.samza.factory", "org.apache.samza.system.kafka.KafkaSystemFactory");
+    configs.put("systems.kafka.consumer.zookeeper.connect", zkConnect());
+    configs.put("systems.kafka.producer.bootstrap.servers", bootstrapUrl());
+    configs.put("systems.kafka.samza.key.serde", "string");
+    configs.put("systems.kafka.samza.msg.serde", "string");
+    configs.put("systems.kafka.samza.offset.default", "oldest");
+    configs.put("job.coordinator.system", "kafka");
+    configs.put("job.default.system", "kafka");
+    configs.put("job.coordinator.replication.factor", "1");
+    configs.put("task.window.ms", "1000");
+
+    if (overriddenConfigs != null) {
+      configs.putAll(overriddenConfigs);
+    }
+
+    app = streamApplication;
+    runner = ApplicationRunner.fromConfig(new MapConfig(configs));
```

Comment:
Where are we specifying the runner to use? Is this using the RemoteApplicationRunner by default?